File: Cpp/1186_Maximum_Subarray_Sum_with_One_Deletion.cpp
class Solution {
public:
    int maximumSum(vector<int>& arr) {
        int deleted = 0;
        int not_deleted = arr[0];
        int result = arr[0];
        
        for(int i=1; i<arr.size(); ++i) {
            deleted = max(not_deleted, deleted + arr[i]);
            not_deleted = max(not_deleted + arr[i], arr[i]);
            result = max({result, deleted, not_deleted});
        }
        
        return result;
    }
};

File: Kotlin/1186_Maximum_Subarray_Sum_with_One_Deletion.kt
class Solution {
    fun maximumSum(arr: IntArray): Int {
        var result = arr[0]
        var deleted = 0
        var notDeleted = arr[0]
        
        for(i in 1 until arr.size) {
            deleted = maxOf(notDeleted, deleted + arr[i])
            notDeleted = maxOf(notDeleted + arr[i], arr[i])
            result = maxOf(result, maxOf(deleted, notDeleted))
        }
        
        return result
    }
}